### ISSUE TITLE
get R CMD check to succeed

### ISSUE THREAD

**Issue by aappling-usgs** [USGS-R/scipiper#88]
Now that David has Travis running, it'd be extra nice for the package to build and check cleanly.

**Comment by aappling-usgs** [comment]
Current Travis build status (e.g., https://travis-ci.org/USGS-R/scipiper/jobs/470080828):
```
Building package
Building with: R CMD build --no-build-vignettes --no-manual --no-vignettes
11.23s$ R CMD build  .
* checking for file ‘./DESCRIPTION’ ... OK
* preparing ‘scipiper’:
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... ERROR
USGS Support Package: https://owi.usgs.gov/R/packages.html#support
Attaching package: 'dplyr'
The following objects are masked from 'package:stats':
    filter, lag
The following objects are masked from 'package:base':
    intersect, setdiff, setequal, union
Quitting from lines 176-179 (task_plans.Rmd) 
Error: processing vignette 'task_plans.Rmd' failed with diagnostics:
All tries are exhausted, but 3 tasks remain
Execution halted
The command "R CMD build  ." failed and exited with 1 during .
Your build has been stopped.
```

Current local `check` status:
```
==> devtools::check(cleanup = FALSE)

Updating scipiper documentation
Loading scipiper
USGS Support Package: https://owi.usgs.gov/R/packages.html#support
Writing NAMESPACE
Writing NAMESPACE
Setting env vars --------------------------------------------------------------
CFLAGS  : -Wall -pedantic
CXXFLAGS: -Wall -pedantic
Building scipiper -------------------------------------------------------------
"C:/PROGRA~1/R/R-35~1.1/bin/x64/R" --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD build "D:\APAData\GitHub\Pipeline\scipiper"  \
  --no-resave-data --no-manual 

* checking for file 'D:\APAData\GitHub\Pipeline\scipiper/DESCRIPTION' ... OK
* preparing 'scipiper':
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... OK
Warning: 'inst/doc' files
    'shared_cache.Rmd', 'task_plans.Rmd', 'shared_cache.html',  'task_plans.html', 'shared_cache.R', 'task_plans.R'
  ignored as vignettes have been rebuilt.
  Run R CMD build with --no-build-vignettes to prevent rebuilding.
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building 'scipiper_0.0.15.tar.gz'

Setting env vars --------------------------------------------------------------
_R_CHECK_CRAN_INCOMING_ : FALSE
_R_CHECK_FORCE_SUGGESTS_: FALSE
Checking scipiper -------------------------------------------------------------
"C:/PROGRA~1/R/R-35~1.1/bin/x64/R" --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD check  \
  "C:\Users\aappling\AppData\Local\Temp\1\Rtmpe2snfn/scipiper_0.0.15.tar.gz"  \
  --as-cran --timings --no-manual 

* using log directory 'D:/APAData/GitHub/Pipeline/scipiper.Rcheck'
* using R version 3.5.1 (2018-07-02)
* using platform: x86_64-w64-mingw32 (64-bit)
* using session charset: ISO8859-1
* using options '--no-manual --as-cran'
* checking for file 'scipiper/DESCRIPTION' ... OK
* this is package 'scipiper' version '0.0.15'
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... NOTE
Found the following hidden files and directories:
  .travis.yml
These were most likely included in error. See section 'Package
structure' in the 'Writing R Extensions' manual.
* checking for portable file names ... OK
* checking serialization versions ... OK
* checking whether package 'scipiper' can be installed ... OK
* checking installed package size ... OK
* checking package directory ... OK
* checking 'build' directory ... OK
* checking DESCRIPTION meta-information ... WARNING
Dependence on R version '3.3.2' not with patchlevel 0
* checking top-level files ... NOTE
Non-standard file/directory found at top level:
  'build.txt'
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking dependencies in R code ... WARNING
'::' or ':::' import not declared from: 'digest'
Unexported objects imported by ':::' calls:
  'remake:::remake' 'remake:::remake_dependency_graph'
  'remake:::remake_is_current' 'remake:::remake_status'
  'remake:::target_is_file'
  See the note in ?`:::` about the use of this operator.
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... NOTE
gd_confirm_posted: no visible global function definition for 'tail'
gd_confirm_posted: no visible binding for global variable
  'drive_resource'
gd_confirm_posted: no visible binding for global variable '.'
gd_get: no visible global function definition for 'tail'
gd_locate_file: no visible binding for global variable 'drive_resource'
gd_locate_file: no visible binding for global variable 'parents'
gd_locate_file: no visible binding for global variable 'name'
gd_put: no visible global function definition for 'tail'
gd_put: no visible binding for global variable '.'
gd_put: no visible binding for global variable 'name'
gd_put: no visible binding for global variable 'drive_resource'
get_relative_path: no visible binding for global variable '.'
hash_targets: no visible global function definition for 'setNames'
retry_patiently: no visible global function definition for 'runif'
s3_confirm_posted: no visible binding for global variable 'Key'
Undefined global functions or variables:
  . Key drive_resource name parents runif setNames tail
Consider adding
  importFrom("stats", "runif", "setNames")
  importFrom("utils", "tail")
to your NAMESPACE file.
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd line widths ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... WARNING
Undocumented arguments in documentation object 'create_task_makefile'
  'ind_complete'

Undocumented arguments in documentation object 'gd_put'
  'dry_put'

Undocumented arguments in documentation object 'loop_tasks'
  'ind_ext' 'verbose'
Documented arguments not in \usage in documentation object 'loop_tasks':
  'vebose'

Undocumented arguments in documentation object 'retry_patiently'
  'verbose'

Undocumented arguments in documentation object 's3_put'
  'dry_put'

Functions with \usage entries need to have the appropriate \alias
entries, and all their arguments documented.
The \usage entries must correspond to syntactically valid R code.
See chapter 'Writing R documentation files' in the 'Writing R
Extensions' manual.
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking installed files from 'inst/doc' ... OK
* checking files in 'vignettes' ... OK
* checking examples ... ERROR
Running examples in 'scipiper-Ex.R' failed
The error most likely occurred in:

> base::assign(".ptime", proc.time(), pos = "CheckExEnv")
> ### Name: as_ind_file
> ### Title: Returns the indicator name corresponding to the given data file
> ###   name
> ### Aliases: as_ind_file
> 
> ### ** Examples
> 
> as_ind_file('mydata.rds') # 'mydata.rds.ind'
[1] "mydata.rds.ind"
> as_ind_file('mydata.rds', ind_ext='st') # 'mydata.rds.st'
[1] "mydata.rds.st"
> as_ind_file('mydata.rds.ind') # Error: "data_file is an indicator file already"
Error in as_ind_file("mydata.rds.ind") : 
  data_file contains indicator files: mydata.rds.ind
Execution halted
* checking for unstated dependencies in 'tests' ... OK
* checking tests ...
  Running 'testthat.R' ERROR
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
  3: scmake("models.ind")
  4: RDSify_build_status(remake_file = remake_file)
  5: remake:::remake(remake_file = remake_file, verbose = FALSE, load_sources = FALSE)
  6: remake_new(remake_file, verbose, load_sources, allow_missing_packages)
  7: read_remake_file(obj$file)
  8: yaml_read(filename)
  9: tryCatch(yaml_load(read_file(filename)), error = catch_yaml)
  10: tryCatchList(expr, classes, parentenv, handlers)
  11: tryCatchOne(expr, names, parentenv, handlers[[1L]])
  12: value[[3L]](cond)
  13: stop(sprintf("while reading '%s'\n%s", filename, e$message), call. = FALSE)
  
  == DONE =======================================================================================================================================
  Error: Test failures
  Execution halted

* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in 'inst/doc' ... OK
* checking re-building of vignette outputs ... WARNING
Error in re-building vignettes:
  ...
Warning in engine$weave(file, quiet = quiet, encoding = enc) :
  The vignette engine knitr::rmarkdown is not available, because the rmarkdown package is not installed. Please install it.
USGS Support Package: https://owi.usgs.gov/R/packages.html#support

Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union

Warning in engine$weave(file, quiet = quiet, encoding = enc) :
  The vignette engine knitr::rmarkdown is not available, because the rmarkdown package is not installed. Please install it.
Quitting from lines 176-179 (task_plans.Rmd) 
Error: processing vignette 'task_plans.Rmd' failed with diagnostics:
All tries are exhausted, but 3 tasks remain
Execution halted

* DONE
Status: 2 ERRORs, 4 WARNINGs, 3 NOTEs

See
  'D:/APAData/GitHub/Pipeline/scipiper.Rcheck/00check.log'
for details.

checking examples ... ERROR
Running examples in 'scipiper-Ex.R' failed
The error most likely occurred in:

> base::assign(".ptime", proc.time(), pos = "CheckExEnv")
> ### Name: as_ind_file
> ### Title: Returns the indicator name corresponding to the given data file
> ###   name
> ### Aliases: as_ind_file
> 
> ### ** Examples
> 
> as_ind_file('mydata.rds') # 'mydata.rds.ind'
[1] "mydata.rds.ind"
> as_ind_file('mydata.rds', ind_ext='st') # 'mydata.rds.st'
[1] "mydata.rds.st"
> as_ind_file('mydata.rds.ind') # Error: "data_file is an indicator file already"
Error in as_ind_file("mydata.rds.ind") : 
  data_file contains indicator files: mydata.rds.ind
Execution halted

checking tests ... ERROR
  Running 'testthat.R'
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
  3: scmake("models.ind")
  4: RDSify_build_status(remake_file = remake_file)
  5: remake:::remake(remake_file = remake_file, verbose = FALSE, load_sources = FALSE)
  6: remake_new(remake_file, verbose, load_sources, allow_missing_packages)
  7: read_remake_file(obj$file)
  8: yaml_read(filename)
  9: tryCatch(yaml_load(read_file(filename)), error = catch_yaml)
  10: tryCatchList(expr, classes, parentenv, handlers)
  11: tryCatchOne(expr, names, parentenv, handlers[[1L]])
  12: value[[3L]](cond)
  13: stop(sprintf("while reading '%s'\n%s", filename, e$message), call. = FALSE)
  
  == DONE =======================================================================================================================================
  Error: Test failures
  Execution halted

checking DESCRIPTION meta-information ... WARNING
Dependence on R version '3.3.2' not with patchlevel 0

checking dependencies in R code ... WARNING
'::' or ':::' import not declared from: 'digest'
Unexported objects imported by ':::' calls:
  'remake:::remake' 'remake:::remake_dependency_graph'
  'remake:::remake_is_current' 'remake:::remake_status'
  'remake:::target_is_file'
  See the note in ?`:::` about the use of this operator.

checking Rd \usage sections ... WARNING
Undocumented arguments in documentation object 'create_task_makefile'
  'ind_complete'

Undocumented arguments in documentation object 'gd_put'
  'dry_put'

Undocumented arguments in documentation object 'loop_tasks'
  'ind_ext' 'verbose'
Documented arguments not in \usage in documentation object 'loop_tasks':
  'vebose'

Undocumented arguments in documentation object 'retry_patiently'
  'verbose'

Undocumented arguments in documentation object 's3_put'
  'dry_put'

Functions with \usage entries need to have the appropriate \alias
entries, and all their arguments documented.
The \usage entries must correspond to syntactically valid R code.
See chapter 'Writing R documentation files' in the 'Writing R
Extensions' manual.

checking re-building of vignette outputs ... WARNING
Error in re-building vignettes:
  ...
Warning in engine$weave(file, quiet = quiet, encoding = enc) :
  The vignette engine knitr::rmarkdown is not available, because the rmarkdown package is not installed. Please install it.
USGS Support Package: https://owi.usgs.gov/R/packages.html#support

Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union

Warning in engine$weave(file, quiet = quiet, encoding = enc) :
  The vignette engine knitr::rmarkdown is not available, because the rmarkdown package is not installed. Please install it.
Quitting from lines 176-179 (task_plans.Rmd) 
Error: processing vignette 'task_plans.Rmd' failed with diagnostics:
All tries are exhausted, but 3 tasks remain
Execution halted


checking for hidden files and directories ... NOTE
Found the following hidden files and directories:
  .travis.yml
These were most likely included in error. See section 'Package
structure' in the 'Writing R Extensions' manual.

checking top-level files ... NOTE
Non-standard file/directory found at top level:
  'build.txt'

checking R code for possible problems ... NOTE
gd_confirm_posted: no visible global R CMD check results
2 errors | 4 warnings | 3 notes
Warning message:
`cleanup` is deprecated 
function definition for 'tail'
gd_confirm_posted: no visible binding for global variable
  'drive_resource'
gd_confirm_posted: no visible binding for global variable '.'
gd_get: no visible global function definition for 'tail'
gd_locate_file: no visible binding for global variable 'drive_resource'
gd_locate_file: no visible binding for global variable 'parents'
gd_locate_file: no visible binding for global variable 'name'
gd_put: no visible global function definition for 'tail'
gd_put: no visible binding for global variable '.'
gd_put: no visible binding for global variable 'name'
gd_put: no visible binding for global variable 'drive_resource'
get_relative_path: no visible binding for global variable '.'
hash_targets: no visible global function definition for 'setNames'
retry_patiently: no visible global function definition for 'runif'
s3_confirm_posted: no visible binding for global variable 'Key'
Undefined global functions or variables:
  . Key drive_resource name parents runif setNames tail
Consider adding
  importFrom("stats", "runif", "setNames")
  importFrom("utils", "tail")
to your NAMESPACE file.

R CMD check succeeded
```

**Comment by aappling-usgs** [comment]
`wontfix`:
```
* checking dependencies in R code ... WARNING
'::' or ':::' import not declared from: 'digest'
Unexported objects imported by ':::' calls:
  'remake:::remake' 'remake:::remake_dependency_graph'
  'remake:::remake_is_current' 'remake:::remake_status'
  'remake:::target_is_file'
  See the note in ?`:::` about the use of this operator.
```
We need those internal functions.

**Comment by aappling-usgs** [comment]
After #89, we're down to this:

Travis (e.g., https://travis-ci.org/USGS-R/scipiper/jobs/470184379):
```
Building package
Building with: R CMD build --no-build-vignettes --no-manual
11.63s$ R CMD build  .
* checking for file ‘./DESCRIPTION’ ... OK
* preparing ‘scipiper’:
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... ERROR
USGS Support Package: https://owi.usgs.gov/R/packages.html#support
Attaching package: 'dplyr'
The following objects are masked from 'package:stats':
    filter, lag
The following objects are masked from 'package:base':
    intersect, setdiff, setequal, union
Quitting from lines 178-181 (task_plans.Rmd) 
Error: processing vignette 'task_plans.Rmd' failed with diagnostics:
All tries are exhausted, but 3 tasks remain
Execution halted
The command "R CMD build  ." failed and exited with 1 during .
Your build has been stopped.
```

plus the clean devtools::test() and the issue
```
Status: 1 WARNING

See
  'D:/APAData/GitHub/Pipeline/scipiper.Rcheck/00check.log'
for details.


checking re-building of vignette outputs ... WARNING
Error in re-building vignettes:
  ...
USGS Support Package: https://owi.usgs.gov/R/packages.html#support

Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union

Quitting from lines 178-181 (task_plans.Rmd) 
Error: processing vignette 'task_plans.Rmd' failed with diagnostics:
All tries are exhausted, but 3 tasks remain
Execution halted
```
in devtools::check() as noted in this PR comment: https://github.com/USGS-R/scipiper/pull/89#issuecomment-448721193

**Comment by aappling-usgs** [comment]
Woo hoo! Got it building! https://travis-ci.org/USGS-R/scipiper/jobs/470677403
![image](https://user-images.githubusercontent.com/12039957/50308696-b108e600-0461-11e9-8d45-1fe578a46ac4.png)
